satpy/composites/__init__.py

```diff
@@ -227,17 +227,20 @@ class SingleBandCompositor(CompositeBase):
     This preserves all the attributes of the dataset it is derived from.
     """
 
+    @staticmethod
+    def _update_missing_metadata(existing_attrs, new_attrs):
+        for key, val in new_attrs.items():
+            if key not in existing_attrs and val is not None:
+                existing_attrs[key] = val
```

Comment:
I see that you check `val is not None`, so that means you check the case when the entry in the dictionary has a None value, but does that actually happen? If not, a simpler solution would be `new_attrs.update(existing_attrs)`...

Reply:
I think this is just a rewrite of the pre-existing version that used dict-comprehension and did the exact same.

Reply:
It is a rewrite with an addition of the `key not in existing_attrs`. If I had to guess the `val is not None` check was there because DataID parameters default to `None` so we (the original author of this code) didn't want a valid DataID value to be overwritten with `None`.

Reply:
With the new DataID objects, we don't default to None anymore, but maybe fixing this is out of the scope of this PR

Reply:
Ah true and good point. I'd rather leave it for now just to be sure.